src/main/java/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java

```diff
@@ -0,0 +1,106 @@
+package internet.shop.dao.jdbc;
+
+import internet.shop.dao.ProductDao;
+import internet.shop.lib.Dao;
+import internet.shop.model.Product;
+import internet.shop.util.ConnectionUtil;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class ProductDaoJdbcImpl implements ProductDao {
+    @Override
+    public Product create(Product object) {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(
+                    "INSERT INTO products(name, price) VALUES (?, ?)",
+                    Statement.RETURN_GENERATED_KEYS);
+            statement.setString(1, object.getName());
+            statement.setDouble(2, object.getPrice());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                object.setId(resultSet.getLong(1));
+            }
+            return object;
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't create product", e);
+        }
+    }
+
+    @Override
+    public Optional<Product> get(Long id) {
+        Product product = null;
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(
+                    "SELECT * FROM products WHERE deleted = 0 AND product_id = ?");
```

Comment:
Not sure about this, but maybe we should have semicolons at the end of the SQL query, like this: `...product_id=?;");` ?

Reply:
Agree

Reply:
Fixed